cmd/list_bundles_test.go

```diff
@@ -1,15 +1,49 @@
 package cmd
 
 import (
+	"log"
+	"os"
+	"os/exec"
+
 	. "github.com/onsi/ginkgo/v2/dsl/core"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("List Bundles Cmd", func() {
 	When("Calling opcap list bundles", func() {
+		// test list bundles '--from-repo' flag
+		It("should succeed", func() {
+			_, err := executeCommand(listBundlesCmd(), createFakeRepo(os.Args[1]))
+			Expect(err).ToNot(HaveOccurred())
+		})
+		// test list bundles cmd '--from-dir' flag
 		It("should succeed", func() {
-			_, err := executeCommand(listBundlesCmd(), []string{}...)
+			_, err := executeCommand(listBundlesCmd(), []string{"--from-dir=../internal/bundle/testrepo/certified-operators"}...)
 			Expect(err).ToNot(HaveOccurred())
 		})
+		// delete fake repo after testing is finished
+		It("should succeed", func() {
+			message := deleteFakeRepo(os.Args[1])
+			Expect(message).To(Equal("testrepo deleted"))
+		})
 	})
 })
+
+// calls create script and creates fake repo containing acc-operator bundles from .test folder for testing
+func createFakeRepo(target string) string {
+	cmd, err := exec.Command("/bin/sh", "../internal/bundle/create_repo.sh", target).Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+	flag := string(cmd)
+	return flag
+}
+
+// calls script and deletes fakerepo folder
+func deleteFakeRepo(target string) string {
+	_, err := exec.Command("/bin/sh", "../internal/bundle/delete_repo.sh", target).Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return "testrepo deleted"
+}
```

Comment:
I believe we could try using the afero file system here. That would discard the need for shell scripts. Here is the afero link https://github.com/spf13/afero I can help you if you need it. We have actually a few examples here in opcap of this.

Reply:
Agree but I would say the afero thing can be done in a separate pr as an improvement.

Reply:
Will create a separate issue as an improvement of current test implementing afero file system instead of using scripts